### PR TITLE
JSON?

### PR DESCRIPTION
**Pull request by Sherokiel**
1) Create file history.json to fix bug with no history file
2) Added message "History is empty" if history is empty
3) Add json encode
4) change history .txt to .json.

### CODE REVIEW

File: Calculator.php [hunk 1]
```diff
@@ -128,6 +130,10 @@ function show_info_block()
 
 Function show_history()
 {
-    $history = file_get_contents('history.txt');
+    $history = file_get_contents('history.json');
+    if($history == null)
+    {
```
Comment (DenTray) [inline Calculator.php]:
```suggestion
    if($history == null) {
```

File: Calculator.php [hunk 2]
```diff
@@ -2,16 +2,22 @@
 require 'constants.php';
 require 'libraries\console_helpers.php';
 
+fopen('history.json', 'a+');
+
+$history = file_get_contents('history.json');
 $times = 0;
-$history = [];
+
+if (!$history == NULL) {
```
Comment (DenTray) [inline Calculator.php]:
```suggestion
if ($history !== null) {
```

File: Calculator.php [hunk 3]
```diff
@@ -2,16 +2,22 @@
...
+    $history = json_decode($history);
```
Comment (DenTray) [inline Calculator.php]:
```suggestion
    $history = json_decode($history, true);
```

File: Calculator.php [hunk 4]
```diff
@@ -28,10 +34,13 @@
 
     info('Result: ' . $result);
     info('=====================');
+    
+    if($history == NULL){
+        $history = [];
+    }
```
Comment (DenTray) [inline Calculator.php]: please move it to the start on the program

File: Calculator.php [hunk 5]
```diff
@@ -28,10 +34,13 @@
...
 
-    $history = "{$argument1} {$command} {$argument2}  =  {$result}" . PHP_EOL;
+    array_push($history,"{$argument1} {$command} {$argument2}  =  {$result}" );
```
Comment (DenTray) [inline Calculator.php]:
```suggestion
    array_push($history, "{$argument1} {$command} {$argument2} = {$result}");
```

File: Calculator.php [hunk 6]
```diff
@@ -94,11 +103,11 @@ function read_operand($message, $command, $isSecondOperand = false)
     return $argument;
 }
 
-function execute_system_command($command)
+function execute_system_command($command,$history)
```
Comment (DenTray) [inline Calculator.php]: please try to use global history variable without passing it to the method

File: Calculator.php [hunk 7]
```diff
@@ -29,9 +35,12 @@
     info('Result: ' . $result);
     info('=====================');
 
-    $history = "{$argument1} {$command} {$argument2}  =  {$result}" . PHP_EOL;
+    if ($history == NULL) {
+        $history = [ ];
+    }
```
Comment (DenTray) [inline Calculator.php]: please move it to the start of the program

File: Calculator.php [hunk 8]
```diff
@@ -29,9 +35,12 @@
...
+
+    array_push($history, "{$argument1} {$command} {$argument2}  =  {$result}" );
```
Comment (DenTray) [inline Calculator.php]:
```suggestion
    array_push($history, "{$argument1} {$command} {$argument2} = {$result}");
```

File: Calculator.php [hunk 9]
```diff
@@ -108,15 +117,22 @@ function execute_system_command($command)
...
     }
 }
 
-function finish_app()
+function finish_app($history)
 {
     $command = choose('Are you sure to wanna quit? Yes/No ', [AGREE, DEGREE]);
 
     if ($command == AGREE) {
+
```
Comment (DenTray) [inline Calculator.php]:
```suggestion
```

File: Calculator.php [hunk 10]
```diff
@@ -108,15 +117,22 @@ function execute_system_command($command)
...
+        if(!$history == NULL) {
+
+        $history = json_encode($history);
+        file_put_contents('history.json', $history);
```
Comment (DenTray) [inline Calculator.php]:
```suggestion
            $history = json_encode($history);
            file_put_contents('history.json', $history);
```

File: Calculator.php [hunk 11]
```diff
@@ -126,8 +142,13 @@ function show_info_block()
     info((implode(' ;  ', AVAILABLE_COMMANDS)) . ' ;');
 }
 
-Function show_history()
+Function show_history($history)
 {
-    $history = file_get_contents('history.txt');
-    info($history);
+
```
Comment (DenTray) [inline Calculator.php]:
```suggestion
```

File: Calculator.php [hunk 12]
```diff
@@ -126,8 +142,13 @@ function show_info_block()
...
+    if ($history == null) {
```
Comment (DenTray) [inline Calculator.php]:
```suggestion
    if ($history === null) {
```

File: Calculator.php [hunk 13]
```diff
@@ -126,8 +142,13 @@ function show_info_block()
...
+        info('You have no history');
+    } else {
+        foreach ($history as $value)
+            info($value);
```
Comment (DenTray) [inline Calculator.php]:
```suggestion
        foreach ($history as $value) {
            info($value);
        }
```

**Review by DenTray — CHANGES_REQUESTED** [review]

File: Calculator.php [hunk 14]
```diff
@@ -2,16 +2,26 @@
...
 require 'libraries\console_helpers.php';
 
+fopen('history.json', 'a+');
+
+$history = file_get_contents('history.json');
 $times = 0;
-$history = [];
+
+if ($history == null) {
+    $history = [ ];
```
Comment (DenTray) [inline Calculator.php]:
```suggestion
    $history = [];
```

File: Calculator.php [hunk 15]
```diff
@@ -2,16 +2,26 @@
...
+}
+
+if ($history != null) {
```
Comment (DenTray) [inline Calculator.php]:
```suggestion
} else {
```

File: Calculator.php [hunk 16]
```diff
@@ -2,16 +2,26 @@
 require 'constants.php';
 require 'libraries\console_helpers.php';
 
+fopen('history.json', 'a+');
+
+$history = file_get_contents('history.json');
 $times = 0;
-$history = [];
+
+if ($history == null) {
```
Comment (DenTray) [inline Calculator.php]:
```suggestion
if ($history === null) {
```

File: Calculator.php (hunk 6 above)
Comment (DenTray) [reply Calculator.php]: not fixed

File: Calculator.php (hunk 9 above)
Comment (DenTray) [reply Calculator.php]: not fixed

**Review by DenTray — CHANGES_REQUESTED** [review]

File: Calculator.php (hunk 9 above)
Comment (DenTray) [reply Calculator.php]: not fixed again

**Review by DenTray — CHANGES_REQUESTED** [review]

File: Calculator.php [hunk 17]
```diff
@@ -126,8 +139,13 @@ function show_info_block()
     info((implode(' ;  ', AVAILABLE_COMMANDS)) . ' ;');
 }
 
-Function show_history()
+Function show_history($history)
```
Comment (DenTray) [inline Calculator.php]:
```suggestion
function show_history($history)
```

File: Calculator.php (hunk 9 above)
Comment (DenTray) [reply Calculator.php]: and again

**Review by DenTray — CHANGES_REQUESTED** [review]

**Review by DenTray — APPROVED** [review]

**Comment by netherpyro** [comment]
Жесть
Ctrl + Alt + L не выучил 😄?

**Pull request merged**